ipamd/ipamd.go

```diff
@@ -234,6 +240,10 @@ func New(k8sapiClient k8sapi.K8SAPIs, eniConfig *eniconfig.ENIConfigController)
 	}
 
 	c.awsClient = client
+	instanceMaxENIs, _ := c.awsClient.GetENILimit()
```

Comment:
Agree, but don't think that moving `instanceMaxENIs, _ := c.awsClient.GetENILimit()` into `getMaxENI()` is a good idea, since we need `aws client` to get instanceMaxENIs and we doesn't have since `getMaxENI()` is not a method.
So we can simple handle the error in `New` function, but I am not sure what should we do in error case.

Reply:
Well, just a little more refactoring should wol, something like this:
```
// getMaxENI returns the maximum number of ENIs to attach to this instance. This is calculated as the lesser of 
// the limit for the instance type and the value configured via the MAX_ENI environment variable. If the value of 
// the environment variable is 0 or less, it will be ignored and the maximum for the instance is returned.
func func (c *IPAMContext) getMaxENI() (int, error) {
	instanceMaxENI, err := c.awsClient.GetENILimit()
	if err != nil {
		return 0, err
	}
	inputStr, found := os.LookupEnv(envMaxENI)
	envMax := defaultMaxENI
	if found {
		if input, err := strconv.Atoi(inputStr); err == nil && input >= 1 {
			log.Debugf("Using MAX_ENI %v", input)
			envMax = input
		}
	}

	if envMax >= 1 && envMax < instanceMaxENI {
		return envMax, nil
	}
	return instanceMaxENI, nil
}
```
Then call it in `nodeInit()` with
```
	log.Debugf("Start node init")
	maxENIs, err := c.getMaxENI()
	if err != nil {
		return err
	}
```
The error returned will be logged in `_main()` as an initialization failure.